epoxy-processor/src/main/java/com/airbnb/epoxy/EpoxyProcessor.java

```diff
@@ -99,8 +92,25 @@ public synchronized void init(ProcessingEnvironment processingEnv) {
     elementUtils = processingEnv.getElementUtils();
     typeUtils = processingEnv.getTypeUtils();
 
+    try {
+      trees = Trees.instance(processingEnv);
+    } catch (IllegalArgumentException ignored) {
+      try {
+        // Get original ProcessingEnvironment from Gradle-wrapped one or KAPT-wrapped one.
+        for (Field field : processingEnv.getClass().getDeclaredFields()) {
+          if (field.getName().equals("delegate") || field.getName().equals("processingEnv")) {
```

Comment:
Does `delegate` correspond to the Gradle wrapper and `processingEnv` the KAPT one? Would be nice to leave a comment.

Reply:
Right, this isn't a public API but it's unlikely to change. Will add a comment

Reply:
ead1284